File: Week9(24_4_24)/differentialeq.py
import numpy as np
from scipy.integrate import odeint
from matplotlib import pyplot as plt

def model(y, t):
    dydt = -y*t+13
    return dydt


y0 = 1
t = np.linspace(0, 5)


y = odeint(model, y0, t)

plt.plot(t, y)
plt.xlabel('Time')
plt.ylabel('y(t)')
plt.title('Solution of dy/dt = -2y')

plt.show()

File: Week9(24_4_24)/differentialeq.m


% Define initial conditions
y0 = [0, 1, 2];

% Define time vector
t = linspace(1, 10);

% Solve the ODE
[t, y] = ode45(@model, t, y0);

% Plot the solution
plot(t, y);
xlabel('Time');
ylabel('y(t)');
title('Solution of dy/dt = (1-y)/(1.95-y) - y/(0.05+y)');
grid on;



